org.eclipse.scanning.api/src/org/eclipse/scanning/api/annotation/scan/DeviceAnnotations.java

```diff
@@ -17,6 +17,61 @@
 import java.util.HashSet;
 import java.util.Set;
 
+/**
+ * Used to annotate methods as participating in a scan.
+ * <p>
+ * It is possible using annotations to have more than one method annotated
+ * which means a super class can declare its implementation as final, requiring a
+ * subclass to define another annotation.
+ * <p>
+ * Services such as IRunnableDeviceService, IPointsGenerator etc may be declared
+ * in the method which we annotate. These will be passed in if am implementation
+ * of them can be found or null with the method still called (which then will often
+ * cause an NPE causing the scan to abort normally).
+ * <p>
+ * If the information about the scan is required (size, shape etc.) then the class
+ * ScanInformation may be received by the annotated method. If the current position is
+ * needed then the IPosition for it should be declared at the pointStart.
+ * <p>
+ * Examples:<p>
+ * <code><pre>
+ * public class Fred implements IScannable {
+ *     &#64;ScanStart
+ *     public final void prepareVoltages() throws Exception {
+ *        ...
+ *     }
+ *     &#64;ScanEnd
+ *     public void dispose() {
+ *        ...
+ *     }
+ *     ...
+ * }
+ * public class Bill extends Fred {
+ *     &#64;ScanStart
+ *     public void moveToNonObstructingLocation(IRunnableDeviceService<?> rservice) throws Exception {
+ *        ...
+ *     }
+ *     &#64;PointStart
+ *     public void checkNextMoveLegal(IPosition pos) throws Exception {
+ *        ...
+ *     }
+ *     &#64;PointStart
+ *     public void notifyPosition(IPosition pos) throws Exception {
+ *        ...
+ *     }
+ *     &#64;PointEnd
+ *     public void deleteLocation() {
+ *        ...
+ *     }
+ *     &#64;Override
+ *     &#64;ScanEnd
+ *     public void dispose() {
+ *        super.dispose();
+ *        ....
+ *     }
+ *}
+ * </pre></code>
+ */
 public class DeviceAnnotations {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add a private constructor to hide the implicit public one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1118)